scylla/src/transport/topology.rs

```diff
@@ -267,7 +268,7 @@ impl Metadata {
                 let token = ((id as u128) << 64) / initial_peers.len() as u128;
 
                 Peer {
-                    address: endpoint.address(),
+                    address: NodeAddr::RealPeer(endpoint.address()),
```

Comment:
I don't think we can treat an initial address as a `RealPeer`. This function will be called with contact point addresses because it is only called on first metadata fetch, so those addresses shouldn't be translated. If address translation is enabled then, in case of a failed first metadata fetch, the session will not work anymore.

The semantics of `FixedLocal` are more appropriate here, though the name isn't. I propose to change the variant names of `NodeAddr` to something more generic - something that only refers to whether the contained address should be translated or not.

By the way, it's too easy to construct a wrong `NodeAddr` from an `UntranslatedEndpoint` via the `.address()` method.

Reply:
I adjusted the names and usage as suggested. However, I can't see any sensible solution the to problem of creating erroneuously wrong `NodeAddr` via `.address()`.

Reply:
Do you have any ideas @piodul?

Reply:
I've rewritten the `address()` method to return `NodeAddr` always. In case of a `ContactPoint`, a `NodeAddr::Untranslatable` is returned.